include/proxy-wasm/context.h

```diff
@@ -142,8 +143,8 @@ class ContextBase : public RootInterface,
   ContextBase();                                                   // Testing.
   ContextBase(WasmBase *wasm);                                     // Vm Context.
   ContextBase(WasmBase *wasm, std::shared_ptr<PluginBase> plugin); // Root Context.
```

Comment:
Do we need to add `std::shared_ptr<PluginHandleBase>` to the Root Context constructor as well?

Reply:
IIUC, root contexts are owned by WasmBase (root_contexts_) and deleted by PluginHandle's destructor
- https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/master/include/proxy-wasm/wasm.h#L193
- https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/master/include/proxy-wasm/wasm.h#L325
- https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/master/src/wasm.cc#L413-L435

and even if they are in pending state, they already outlive WasmBase (https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/master/include/proxy-wasm/wasm.h#L292-L294).

So that's the reason why I intentionally didn't pass handles to roots.. any thoughts?